### ISSUE TITLE
Defining at least one zooplankton functional group for each OSMOSE case study

### ISSUE THREAD

**Issue by agruss2** [jhpoelen/fb-osmose-bridge#120]
@FIN-JBarile @FIN-casey @jhpoelen
At least one zooplankton functional group should be defined for each OSMOSE case study.
Zooplankton species are included in SeaLifeBase. However, their "data richness" - as defined in #59 - is often very low. 
I would like: (1) the bridge between FishBase/SeaLifeBase and OSMOSE to define at least one zooplankton functional group for each OSMOSE case study; and (2) the zooplankton group(s) defined to have a maximum of 30 species.

Here are some comments:
(1) Since "data richness" is often very low for zooplankton species, here are the default values that should be defined **for zooplankton species only**:
(i) Minimum size: 0.002 cm (OSMOSE parameter: “plankton.size.min.plkX”);
(ii) Maximum size: 0.3 cm (OSMOSE parameter: “plankton.size.max.plkX”);
(iii) Trophic level: 2 (OSMOSE parameter: "plankton.TL.plkX").
(2) If the user selects an ecosystem or a country plus a FAO area (say, the "Iceland Shelf/Sea" ecosystem) and there is no zooplankton species for the ecosystem or country plus FAO area of interest (e.g., the "Iceland Shelf/Sea" ecosystem), then the UI should still create a "zooplankton" background functional group in the Table that is provided in Step 2, with the columns "Species" and "Class" both empty. Then, the bridge between FishBase/SeaLifeBase and OSMOSE will provide the following parameter values for the "zooplankton" group that it created:
(i) Minimum size: 0.002 cm;
(ii) Maximum size: 0.3 cm;
(iii) Trophic level: 2.
**And in this specific case (i.e., when there is no zooplankton species for the ecosystem or country plus FAO area of interest in SeaLifeBase)**, if the user attempts to redefine the zooplankton background functional group created by the bridge between FishBase/SeaLifeBase and OSMOSE as a "focal" functional group, then a warning message should pop up; this warning message should say: "The Zooplankton group can only be defined as a background functional group". 

Could you please implement all of the above? Please let me know if you have any questions and/or comments. Many thanks.

**Comment by agruss2** [comment]
@FIN-casey @FIN-JBarile Please implement what is described above; and let me know when you think this issue has been solved so that I can run a new test with the bridge between FishBase/SeaLifeBase and OSMOSE. Many thanks.

**Comment by FIN-casey** [comment]
I implemented this issue. If there is no "zooplankton" functional group in the table, the UI will add the "zooplankton" functional group with no classes and species. Since the group is required for each OSMOSE study, I put an alert of "The functional group cannot be deselected." whenever the user tries to deselect the functional group. I also put an alert of "The Zooplankton group can only be defined as a background functional group", when the user tries to switch the group to focal functional group.

Please verify.

**Comment by agruss2** [comment]
@FIN-casey @FIN-JBarile  This is wonderful, many thanks!
I have three comments for you. 
First, I have a very minor comment for you:
Could you please replace "The functional group cannot be deselected." with: "The zooplankton group cannot be deselected."?

**Comment by agruss2** [comment]
@FIN-casey @FIN-JBarile 
My second comment relates to the "osm_param-ltl.csv" file. 
I just ran a test where I queried parameters for an OSMOSE model for the Iceland Shelf/Sea ecosystem. 
I did not touch anything in Step 2 and downloaded the config zip file. 
However, when I looked at the "osm_param-ltl.csv" file (provided here in the .xlsx format: 
[osm_param-ltl.xlsx](https://github.com/jhpoelen/fb-osmose-bridge/files/1071611/osm_param-ltl.xlsx)), the parameter values for the zooplankton group were all "NA", while they should be as follows:
 (i) Minimum size: 0.002 cm (OSMOSE parameter: “plankton.size.min.plkX”);
(ii) Maximum size: 0.3 cm (OSMOSE parameter: “plankton.size.max.plkX”);
(iii) Trophic level: 2 (OSMOSE parameter: "plankton.TL.plkX").

Could you please correct for this?

**Comment by agruss2** [comment]
@FIN-casey @FIN-JBarile @Dengaloo 
I have a third comment. 
I ran another test with the bridge between FishBase/SeaLifeBase and OSMOSE, where I queried parameters for an OSMOSE model for the Iceland Shelf/Sea ecosystem. I did not touch anything in Step 2, except that I defined the "invertpelagicsm" functional group as a "background" functional group. The "invertpelagicsm" functional group includes only one species: _Calanus finmarchicus_ (Maxillopoda); see: 
[functional_groups.xlsx](https://github.com/jhpoelen/fb-osmose-bridge/files/1071665/functional_groups.xlsx)
However, as you know, _Calanus finmarchicus_ belongs to the zooplankton. Like pretty much all the zooplankton species included in SeaLifeBase, _Calanus finmarchicus_ is a data-poor species and, therefore, the bridge between FishBase/SeaLifeBase and OSMOSE provides only NA's for this species. 
Therefore, I think that we should find a way to: (1) make the bridge between FishBase/SeaLifeBase and OSMOSE define only general "phytoplankton" and "zooplankton" background functional groups, as is presently the case; and (2) prevent the bridge from defining other phytoplankton and zooplankton groups/species in the table provided in Step 2.
 
This may be a little bit technical to implement. 

Is there a way to identify which of the numerous species included in SeaLifeBase belong to the phytoplankton or zooplankton? 
Or are we only able to determine the class, phylum, order, etc. to which the species that are included in SeaLifeBase belong? 

In any case, it would be good to implement what I described above, so as to avoid defining redundant plankton groups for which the bridge between FishBase/SeaLifeBase and OSMOSE will only provide NA's.
